### ISSUE TITLE
Feature request/ question about git diff --name-only --relative $(git merge-base ".. branchName .." HEAD

### ISSUE THREAD

**Issue by arnevm123** [aaronhallaert/advanced-git-search.nvim#48]

**Comment by aaronhallaert** [comment]
Yes, it has the same purpose. The only difference is that I want it to be more general, where the base branch is not `main` or `master`. For example, when you have a feature branch A based on feature branch B, I only want to see changes since the creation of feature branch A. I also noticed it does not always work if I have multiple local branches, that's why I marked it as "experimental" in the README. It's something that needs some investigation, feel free to open a PR 🚀 .  If possible, you can extend/fix the current implementation of `changed_on_branch`. I'll have a look at the current implementation, and try to reproduce the cases where it doesn't work

**Comment by aaronhallaert** [comment]
Maybe we can first investigate why it doesn't work for you.

The current implementation uses the following command to determine the base branch:
```sh
git show-branch | \
    sed "s/].*//" | \
    grep "*" | \
    grep -v "$(git rev-parse --abbrev-ref HEAD)" | \
    head -n1 | \
    sed "s/^.*\\[//"
```
Can you try this command in the terminal where the picker is not working?

Then this command is used to list the changed files.
https://github.com/aaronhallaert/advanced-git-search.nvim/blob/5637d80861e253a8621450eb58621009e1307deb/lua/advanced_git_search/commands/find.lua#L111-L123

The picker will only show changes that are added to the git index.

**Comment by arnevm123** [comment]
I get an error saying "cannot handle more than 26 refs"
So I guess there's too many branches on the repo I'm running it on.
I can do some more investigating tonight after my day job.

**Comment by arnevm123** [comment]
After some investigation, I came across several potential solutions here: https://stackoverflow.com/questions/3161204/how-to-find-the-nearest-parent-of-a-git-branch

However, none of the solutions I tested were effective in all scenarios. Some solutions failed when dealing with remote branches, while others didn't handle branches merged with the master branch correctly.

Among the available options, the following solution appears to be the most promising: 
https://stackoverflow.com/questions/3161204/how-to-find-the-nearest-parent-of-a-git-branch/68673744#68673744. 

Once I have more time, I intend to delve deeper into this solution and explore its feasibility further.

**Comment by aaronhallaert** [comment]
Yeah, it's not an easy problem to solve. I have some difficulties figuring this out as well. Thanks for taking the time to investigate this further. I think it's ok if we first focus on local branches.

FYI: My current implementation which relies on `git show-branch` also doesn't work if the base of the feature branch is some commits behind the HEAD of `main` (for example the base commit is `main~4`). In some cases, `git show-branch` will not show the commits as `main~4`, `main~5`, `main~6`, but as `feature-branch~1`, `feature-branch~2`, `feature-branch~3`. All the entries which contain the `feature-branch` name are filtered, so in this case, the command won't return anything.

**Comment by arnevm123** [comment]
What if, instead of having the command hard-coded, we provide an option for users to customize the behavior by allowing them to add their own script or hardcoded string during the setup process? 

In this approach, we can use your command as the default option, but also allow users to modify it according to their specific needs.

I can look into the implementation if this is okay for you 😄

**Comment by aaronhallaert** [comment]
Maybe a better solution could be to have an initial picker to select a local branch. Then, the merge base can be determined for the selected branch and the current branch with `git merge-base`. Eventually, we can show the current picker that lists the changed files.

I'd rather have it more explicit by letting the user pick the branch than letting them specify a complex command in the configs.

**Comment by arnevm123** [comment]
That might be a better resolution! I'll see if I can find some time to make a PR for this 😄
